app/controllers/media_objects_controller.rb

```diff
@@ -296,27 +296,34 @@ def update_status
     success_count = 0
     Array(params[:id]).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) (permission denied)."]
-      else
-        case status
-          when 'publish'
-            media_object.publish!(user_key)
-            # additional save to set permalink
-            media_object.save( validate: false )
-            success_count += 1
-          when 'unpublish'
-            if can? :unpublish, media_object
-              media_object.publish!(nil)
-              success_count += 1
-            else
-              errors += ["#{media_object.title} (#{id}) (permission denied)."]
-            end
+      success = false
+      case status
```

Comment:
Would clean this logic up as well. Extremely complicated. Could probably just do all this in a simple `if/elsif/else` statement?